modules/EVSE/Evse15118D20/charger/ISO15118_chargerImpl.cpp

```diff
@@ -552,6 +884,58 @@ iso15118::session::feedback::Callbacks ISO15118_chargerImpl::create_callbacks()
         publish_dc_ev_maximum_limits({max_limits.current, max_limits.power, max_limits.voltage});
     };
 
+    callbacks.ac_limits = [this](const feedback::AcLimits& limits) {
+        if (const auto* ac_transfer_mode = std::get_if<dt::AC_CPDReqEnergyTransferMode>(&limits)) {
+            publish_ac_ev_power_limits(fill_ac_ev_power_limits(*ac_transfer_mode));
+        } else if (const auto* ac_bpt_transfer_mode = std::get_if<dt::BPT_AC_CPDReqEnergyTransferMode>(&limits)) {
+            publish_ac_ev_power_limits(fill_ac_ev_power_limits(*ac_bpt_transfer_mode));
+        }
+    };
+
+    callbacks.ac_charge_loop_req = [this](const feedback::AcChargeLoopReq& ac_charge_loop_req) {
+        if (const auto* ac_control_mode = std::get_if<feedback::AcReqControlMode>(&ac_charge_loop_req)) {
+            if (const auto* scheduled_mode = std::get_if<ScheduleControlModeAC>(ac_control_mode)) {
+                if (scheduled_mode->max_charge_power.has_value() or scheduled_mode->max_charge_power_L2.has_value() or
+                    scheduled_mode->max_charge_power_L3.has_value() or scheduled_mode->min_charge_power.has_value() or
+                    scheduled_mode->min_charge_power_L2.has_value() or
+                    scheduled_mode->min_charge_power_L3.has_value()) {
+                    publish_ac_ev_power_limits(fill_ac_ev_power_limits(*scheduled_mode));
+                }
+                publish_ac_ev_present_powers(fill_ac_ev_present_power_values(*scheduled_mode));
+                // TODO(SL): publish_scheduled_control_mode
+            } else if (const auto* bpt_scheduled_mode = std::get_if<BPT_ScheduleReqControlModeAC>(ac_control_mode)) {
+                if (scheduled_mode->max_charge_power.has_value() or scheduled_mode->max_charge_power_L2.has_value() or
+                    scheduled_mode->max_charge_power_L3.has_value() or scheduled_mode->min_charge_power.has_value() or
+                    scheduled_mode->min_charge_power_L2.has_value() or
+                    scheduled_mode->min_charge_power_L3.has_value()) {
+                    publish_ac_ev_power_limits(fill_ac_ev_power_limits(*scheduled_mode));
```

Comment:
Suggestion is already reformatted using clang-format
```suggestion
            } else if (const auto* bpt_scheduled_mode = std::get_if<BPT_ScheduleReqControlModeAC>(ac_control_mode)) {
                if (bpt_scheduled_mode->max_charge_power.has_value() or
                    bpt_scheduled_mode->max_charge_power_L2.has_value() or
                    bpt_scheduled_mode->max_charge_power_L3.has_value() or
                    bpt_scheduled_mode->min_charge_power.has_value() or
                    bpt_scheduled_mode->min_charge_power_L2.has_value() or
                    bpt_scheduled_mode->min_charge_power_L3.has_value()) {
                    publish_ac_ev_power_limits(fill_ac_ev_power_limits(*bpt_scheduled_mode));
```